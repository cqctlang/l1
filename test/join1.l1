


@include <util.cqct>

lst = ["a","b","c","d","e","f","g"];
printf("%s\n",join(lst));
printf("%s\n",join(lst,","));
printf("%s\n",join(lst,", "));

x = "asdfasdkjf;sldkfj;asldkfjsdk;lfjsadfsa;lkdjf";
lst = split(x,"j");
printf("%d\n",x == join(lst,"j"));

printf("%s\n",join([],""));
printf("%s\n",join([],"a"));
printf("%s\n",join(["a","","","","b"],"sdfasdfasdfasdfasdfasdfasd"));

applyk(join,
    @lambda(r) { r; },
    @lambda(m) { printf("%s\n",m); },
    []);
applyk(join,
    @lambda(r) { r; },
    @lambda(m) { printf("%s\n",m); },
    [1]);
applyk(join,
    @lambda(r) { r; },
    @lambda(m) { printf("%s\n",m); },
    [["a"],"b","c"]);
applyk(join,
    @lambda(r) { r; },
    @lambda(m) { printf("%s\n",m); },
    [[1],"c"]);
applyk(join,
    @lambda(r) { r; },
    @lambda(m) { printf("%s\n",m); },
    [["a"],1]);

