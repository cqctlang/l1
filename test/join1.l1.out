abcdefg
a,b,c,d,e,f,g
a, b, c, d, e, f, g
1


asdfasdfasdfasdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdb
not enough arguments to join (expects at least 1, passed 0)
first argument to join must be a list
too many arguments to join
first argument to join must only contain strings
second argument to join must be a string
